### PR TITLE
feat: gossipclient + failover

### PR DESCRIPTION
**Pull request by alanshaw**
merged https://github.com/drand/drand/pull/357 and https://github.com/drand/drand/pull/363 plus some fixes and features

### CODE REVIEW

File: client/failover.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+import (
+	"context"
+	"time"
+
+	"github.com/drand/drand/beacon"
+	"github.com/drand/drand/key"
+	"github.com/drand/drand/log"
+)
+
+const defaultFailoverGracePeriod = time.Second * 5
```
Comment (willscott) [inline client/failover.go:12]: worth defaulting this in relation to the group period?

File: client/failover.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+}
+
+// Watch returns new randomness as it becomes available.
+func (c *failoverWatcher) Watch(ctx context.Context) <-chan Result {
+	var latest uint64
+	ch := make(chan Result, 5)
+
+	sendResult := func(r Result) {
+		if latest >= r.Round() {
+			return
```
Comment (willscott) [inline client/failover.go:43]: worth a log.debug of skipping duplicates

File: client/failover.go (hunk 1 above)
Comment (alanshaw) [reply client/failover.go:12]: https://github.com/drand/drand/pull/383

File: client/failover.go (hunk 2 above)
Comment (alanshaw) [reply client/failover.go:43]: this was added to https://github.com/drand/drand/pull/363

**Comment by alanshaw** [comment]
closing in favour of https://github.com/drand/drand/pull/472

**Pull request closed without merging**